remme/tp/pub_key.py

```diff
@@ -81,26 +118,27 @@ def _store_pub_key(self, context, signer_pubkey, transaction_payload):
             LOGGER.debug(f'entity_hash {transaction_payload.entity_hash}')
             raise InvalidTransaction('Entity hash or signature not a hex format')
 
-        # FIXME: For support PKCS1v15 and PSS
-        LOGGER.warn('HAZARD: Detecting padding for verification')
-        sigerr = 0
-        pkcs = padding.PKCS1v15()
-        pss = padding.PSS(mgf=padding.MGF1(hashes.SHA512()), salt_length=padding.PSS.MAX_LENGTH)
-        for _padding in (pkcs, pss):
-            try:
-                cert_signer_pubkey.verify(ehs_bytes, eh_bytes, _padding, hashes.SHA512())
-                LOGGER.warn('HAZARD: Padding found: %s', _padding.name)
-            except InvalidSignature:
-                sigerr += 1
-
-        if sigerr == 2:
+        try:
+            if transaction_payload.public_key_type == NewPubKeyPayload.RSA:
```

Comment:
We don't need a separate enum here

https://cryptography.io/en/latest/x509/reference/#cryptography.x509.Certificate.public_key
https://cryptography.io/en/latest/x509/reference/#cryptography.x509.Certificate.signature_algorithm_oid

Reply:
@eugene-babichenko it checks on an object, but not from the string, why we need to create an object first then check?

Reply:
Did a more thorough review. Currently it's ok as there is no single format for all keys. Will return to this later.